applications/crossbar/src/modules/cb_rate_limits.erl

```diff
@@ -0,0 +1,231 @@
+%%%-------------------------------------------------------------------
+%%% @copyright (C) 2011-2015, 2600Hz INC
+%%% @doc
+%%%
+%%% Rate limits
+%%% /accounts/{account_id}/rate_limits - manip account's access lists
+%%% /accounts/{account_id}/devices/{device_id}/rate_limits - manip device's access lists
+%%%
+%%% @end
+%%% @contributors:
+%%%   SIPLABS, LLC (Maksim Krzhemenevskiy)
+%%%-------------------------------------------------------------------
+-module(cb_rate_limits).
+
+-export([init/0
+         ,allowed_methods/0
+         ,resource_exists/0
+         ,validate/1
+         ,post/1
+         ,delete/1
+        ]).
+
+-include("../crossbar.hrl").
+
+-define(LISTING_BY_OWNER, <<"rate_limits/list_by_owner">>).
+
+%%%===================================================================
+%%% API
+%%%===================================================================
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Initializes the bindings this module will respond to.
+%% @end
+%%--------------------------------------------------------------------
+-spec init() -> 'ok'.
+init() ->
+    _ = crossbar_bindings:bind(<<"*.allowed_methods.rate_limits">>, ?MODULE, 'allowed_methods'),
+    _ = crossbar_bindings:bind(<<"*.resource_exists.rate_limits">>, ?MODULE, 'resource_exists'),
+    _ = crossbar_bindings:bind(<<"*.validate.rate_limits">>, ?MODULE, 'validate'),
+    _ = crossbar_bindings:bind(<<"*.execute.post.rate_limits">>, ?MODULE, 'post'),
+    _ = crossbar_bindings:bind(<<"*.execute.delete.rate_limits">>, ?MODULE, 'delete').
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Given the path tokens related to this module, what HTTP methods are
+%% going to be responded to.
+%% @end
+%%--------------------------------------------------------------------
+-spec allowed_methods() -> http_methods().
+allowed_methods() ->
+    [?HTTP_GET, ?HTTP_POST, ?HTTP_DELETE].
+
+%%--------------------------------------------------------------------
+%% @public
+%% @doc
+%% Does the path point to a valid resource
+%% So /rate_limits => []
+%%    /rate_limits/foo => [<<"foo">>]
+%%    /rate_limits/foo/bar => [<<"foo">>, <<"bar">>]
+%% @end
+%%--------------------------------------------------------------------
+-spec resource_exists() -> 'true'.
+resource_exists() -> 'true'.
+
+-spec only_if_reseller_or_master(cb_context:context()
+                                 ,fun((cb_context:context()) -> cb_context:context())
+                                ) -> cb_context:context().
+only_if_reseller_or_master(Context, Continue) ->
```

Comment:
I think this should be part of an `authorize` callback. Check against req_nouns to see if it should authorize, then do this check.